### PR TITLE
Last (hopefully) breaking changes ahead of beta

### PR DESCRIPTION
**Pull request by cdata**
This change implements the remaining breaking changes to the Noosphere core data structures that we hope to land ahead of the upcoming Private Beta phase of Subconscious. The breaking changes are several, and I'll attempt to document them here:

## Renames

The following core domain concepts have been renamed to be more coherent or consistent with the way we have come to talk about them among ourselves:

 - Links -> Content
 - Names -> AddressBook / Identities
 - Authorization -> Authority
 - Allowed UCANs -> Delegations
 - Revoked UCANs -> Revocations

## HAMT layout

The HAMT layout changes implemented in #284 have been almost entirely reverted. With the landing of the typed CID in #297 , we can now leverage the type system to be precise about what means when the key or the value of the HAMT is a CID. This revised approach does lead us to have to be more fiddle-y when crawling the sphere to stream its blocks, but otherwise it gives us improved data layout flexibility while maintaining the benefits of the change made in #284 .

## Other breaking changes

The following additional breaking changes were made:

 - More method names that were prefixed with `try_` have had that prefix removed to align with our convention elsewhere
 - Versions of the `fastcdc` and `base64` crates were updated by breaking increments

### CODE REVIEW

File: rust/noosphere-core/src/view/authority.rs [hunk 1]
```diff
@@ -66,15 +66,15 @@ where
         })
     }
 
-    pub async fn try_get_allowed_ucans(&self) -> Result<AllowedUcans<S>> {
+    pub async fn get_delegations(&self) -> Result<Delegations<S>> {
```
Comment (jsantell) [inline rust/noosphere-core/src/view/authority.rs:69]: `delegations` much better than `allowed_ucans`, nice!

File: rust/noosphere-core/src/view/mutation.rs [hunk 2]
```diff
@@ -47,20 +50,20 @@ impl<S: BlockStore> SphereRevision<S> {
...
 pub struct SphereMutation {
     did: String,
-    links: LinksMutation,
-    names: NamesMutation,
-    allowed_ucans: AllowedUcansMutation,
-    revoked_ucans: RevokedUcansMutation,
+    content: ContentMutation,
+    identities: IdentitiesMutation,
+    delegations: DelegationsMutation,
+    revocations: RevocationsMutation,
```
Comment (jsantell) [inline rust/noosphere-core/src/view/mutation.rs:56]: Simple, and given the data spheres store, non-ambiguous, especially compared to "links"/"names" -- huge improvement in clarity here IMO

File: rust/noosphere-core/src/data/address.rs [hunk 3]
```diff
@@ -1,91 +1,133 @@
...
-                            None => {
-                                warn!("No 'link' field in fact, skipping...");
+/// A [LinkRecord] is a newtype that represents a JWT that ought to contain a
+/// [Cid] reference to a sphere
+#[derive(Debug, Eq, PartialEq, Clone, Serialize, Deserialize, Hash)]
+#[serde(from = "Jwt", into = "Jwt")]
+#[repr(transparent)]
+pub struct LinkRecord(Jwt);
+
+impl LinkRecord {
```
Comment (jsantell) [inline rust/noosphere-core/src/data/address.rs:65]: As a "core type", wonder if `noosphere_ns::NsRecord` (or renamed to `noosphere_ns::LinkRecord`) could be replaced by this, or sharing some functionality.

**Review by jsantell — APPROVED** [review]
Fantastic improvements and clarifications here @cdata!

File: rust/noosphere-core/src/data/address.rs (hunk 3 above)
Comment (cdata) [reply rust/noosphere-core/src/data/address.rs:65]: Yah, definitely possible!

**Pull request merged**